File: src/com/spdqbr/aoc/y2025/Day02.java
package com.spdqbr.aoc.y2025;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

import com.spdqbr.aoc.Solution;

public class Day02 extends Solution {
	public static final int day = 2;
	public static final int year = 2025;
	
	
	public static void main(String[] args) {
//		DEBUG = true;
		(new Day02()).main(year, day, 0);
		(new Day02()).main(year, day, 1);
		(new Day02()).main(year, day, 2);
		(new Day02()).main(year, day, 3);
	}
	
	public boolean isValid(long n, Pattern p) {
		String nstr = Long.toString(n);
		Matcher m = p.matcher(nstr);
		return ! m.find();
	}
	
	public String part1(String[] input) {
		Pattern p = Pattern.compile("^(.+)\\1$");
		String[] ranges = input[0].split(",");
		long sum = 0;
		for(String range :  ranges) {
			long lower = Long.parseLong(range.split("-")[0]);
			long upper = Long.parseLong(range.split("-")[1]);
			
			for(long i = lower; i <= upper; i++) {
				if( ! isValid(i, p)) {
					sum += i;
				}
			}
		}
		return ""+sum;
	}
	
	public String part2(String[] input) {
		Pattern p = Pattern.compile("^(.+)\\1+$");
		String[] ranges = input[0].split(",");
		long sum = 0;
		for(String range :  ranges) {
			long lower = Long.parseLong(range.split("-")[0]);
			long upper = Long.parseLong(range.split("-")[1]);
			
			for(long i = lower; i <= upper; i++) {
				if( ! isValid(i, p)) {
					sum += i;
				}
			}
		}
		return ""+sum;
	}
}

File: python/2025/src/Day02.py
from Solution import Solution
import re

part : int = 4

class TodaysSolution(Solution):
    def part_1(self, data: list[str]) -> None:
        data = data[0].split(",")
        pattern: Pattern = re.compile(r'^(.+)\1$')
        sum: int = 0

        for datum in data:
            left, right = datum.split("-")
            for i in range(int(left), int(right)+1):
                if pattern.match(str(i)):
                    sum += i
        print(f"Sum: {sum}")

    def part_2(self, data: list[str]) -> None:
        data = data[0].split(",")
        pattern: Pattern = re.compile(r'^(.+)\1+$')
        sum: int = 0

        for datum in data:
            left, right = datum.split("-")
            for i in range(int(left), int(right)+1):
                if pattern.match(str(i)):
                    sum += i
        print(f"Sum: {sum}")
    
def main(filename: str) -> None:
    today: TodaysSolution = TodaysSolution()
    today.solve(filename, part)

if __name__ == '__main__':
    main(__file__)